### ISSUE TITLE
`ifdef` warnings during  preprocessing

### ISSUE THREAD

**Issue by ThePrez** [IBM/JTOpen#24]
Running Maven builds (for instance `mvn package --file pom-dist.xml` shows a whole bunch of warnings about incorrect ifdefs. For instance: 

```pascal
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionRedirect.java:2485
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionRedirect.java:2501
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionRedirect.java:2517
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionRedirect.java:2532
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:20
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:26
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:30
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:40
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:73
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:1671
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:1723
Warning.  Incorrect ifndef line '/* ifndef JDBC40 ' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:1726
Warning.  Incorrect ifdef line '/* ifdef JDBC40 */' at com/ibm/as400/access/AS400JDBCConnectionHandle.java:1730
```

**Comment by ThePrez** [comment]
@jeber-ibm , what is incorrect about these lines? 
For instance, `com/ibm/as400/access/AS400JDBCConnectionHandle.java:20` and context:
```java

package com.ibm.as400.access;

import javax.sql.ConnectionEvent;
import java.sql.Array;
import java.sql.Blob;
import java.sql.CallableStatement;
/* ifdef JDBC40 */
import java.sql.SQLClientInfoException;
/* endif */ 
import java.sql.Clob;
import java.sql.Connection;        //@A5A
```

Reading through the code, for instance [here](https://github.com/IBM/JTOpen/blob/7fa346cbf233d364e3b26ce3793df7ab6a439b21/build/Jdbc40Task.java#L121), looks like it assumes the "base" code is pre-JDBC40, which is no longer the case. Is that a correct analysis? 

I think we want to tweak the logic in those `Jdbc*Task` classes to account for this. Of course, it's just noise, no functional side effect, so this is low priority

**Comment by jeber-ibm** [comment]
Yes, it assumes that the code is pre JDBC40.           Aren’t we removing the JDBC40 ifdefs since they are not needed any more?


From: Jesse Gorzinski ***@***.***>
Sent: Friday, April 7, 2023 3:25 PM
To: IBM/JTOpen ***@***.***>
Cc: John Eberhard ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [IBM/JTOpen] `ifdef` warnings during preprocessing (Issue #24)

@jeber-ibm , what is incorrect about these lines? For instance, com/ibm/as400/access/AS400JDBCConnectionHandle. java: 20 and context: package com. ibm. as400. access; import javax. sql. ConnectionEvent; import java. sql. Array; import java. sql. Blob;
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

@jeber-ibm<https://github.com/jeber-ibm> , what is incorrect about these lines?
For instance, com/ibm/as400/access/AS400JDBCConnectionHandle.java:20 and context:

package com.ibm.as400.access;



import javax.sql.ConnectionEvent;

import java.sql.Array;

import java.sql.Blob;

import java.sql.CallableStatement;

/* ifdef JDBC40 */

import java.sql.SQLClientInfoException;

/* endif */

import java.sql.Clob;

import java.sql.Connection;        ***@***.***

Reading through the code, for instance here<https://github.com/IBM/JTOpen/blob/7fa346cbf233d364e3b26ce3793df7ab6a439b21/build/Jdbc40Task.java#L121>, looks like it assumes the "base" code is pre-JDBC40, which is no longer the case. Is that a correct analysis?

I think we want to tweak the logic in those Jdbc*Task classes to account for this. Of course, it's just noise, no functional side effect, so this is low priority

**Comment by ThePrez** [comment]
> Yes, it assumes that the code is pre JDBC40.           Aren’t we removing the JDBC40 ifdefs since they are not needed any more?
> 
> 
> 

Yes. Sorry. That makes the most sense. Good first issue for @Afzal50
